src/main/scala/strawman/collection/mutable/HashMap.scala

```diff
@@ -94,11 +94,7 @@ object HashMap extends MapFactory[HashMap] {
 
   def empty[K, V]: HashMap[K, V] = new HashMap[K, V]
 
-  def fromIterable[K, V](it: collection.Iterable[(K, V)]): HashMap[K, V] =
-    it match {
-      case hm: HashMap[K, V] => hm
-      case _ => empty ++= it
-    }
+  def fromIterable[K, V](it: collection.Iterable[(K, V)]): HashMap[K, V] = Growable.fromIterable(empty[K, V], it)
```

Comment:
Note that the type test (the `case hm: HashMap[K, V] => hm` line above) has been removed. This test is needed to prevent infinite recursion in the case of immutable collections because the implementation of `fromIterable` might rely on `fromIterable`. I don’t think this can be a problem for mutable collections.